From: Chad Monroe <chad@monroe.io>
Date: Mon, 08 May 2023 22:07:22 -0700
Subject: [PATCH] net: dsa: mt7530: populate dsa slave of_node 

When creating the MT7530 MDIO bus, populate DSA slave of_node pointer
correctly so external PHY drivers can use it if needed.

Signed-off-by: Chad Monroe <chad@monroe.io>
---
 drivers/net/dsa/mt7530.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/dsa/mt7530.c
+++ b/drivers/net/dsa/mt7530.c
@@ -2289,6 +2289,9 @@ mt7530_setup_mdio(struct mt7530_priv *pr
 	bus->parent = dev;
 	bus->phy_mask = ~ds->phys_mii_mask;
 
+	/* fill out of_node for external PHY drivers if present */
+	ds->slave_mii_bus->dev.of_node = of_get_child_by_name(priv->dev->of_node, "mdio");
+
 	if (priv->irq)
 		mt7530_setup_mdio_irq(priv);
 
